test/config_parser_SUITE.erl

```diff
@@ -1208,71 +1224,81 @@ pool_http_tls(_Config) ->
     ?err(pool_conn_raw(<<"http">>, #{<<"tls">> => <<"secure">>})).
 
 pool_redis_host(_Config) ->
-    ?cfg(pool_config({redis, global, default, [], [{host, "localhost"}]}),
+    ?cfg(pool_config(#{type => redis, scope => global, tag => default, opts => #{},
+                       conn_opts => #{host => "localhost"}}),
          pool_conn_raw(<<"redis">>, #{<<"host">> => <<"localhost">>})),
     ?err(pool_conn_raw(<<"redis">>, #{<<"host">> => 8443})),
     ?err(pool_conn_raw(<<"redis">>, #{<<"host">> => ""})).
 
 pool_redis_port(_Config) ->
-    ?cfg(pool_config({redis, global, default, [], [{port, 6379}]}),
+    ?cfg(pool_config(#{type => redis, scope => global, tag => default, opts => #{},
+                       conn_opts => #{port => 6379}}),
          pool_conn_raw(<<"redis">>, #{<<"port">> => 6379})),
```

Comment:
For each default there should be a test that they work as expected. We could use the `merge_with_defaults` logic instead of specifying them explicitly:
```suggestion
    ?cfg(pool_config(#{type => redis, tag => default}), pool_conn_raw(<<"redis">>, #{})),
    ?cfg(pool_config(#{type => redis, tag => default, conn_opts => #{port => 6333}}),
         pool_conn_raw(<<"redis">>, #{<<"port">> => 6333})), % test something else than the default
```